Framework/PythonInterface/plugins/algorithms/ConvertWANDSCDtoQ.py

```diff
@@ -0,0 +1,319 @@
+        self.declareProperty('NormaliseBy', 'Monitor', StringListValidator(['None', 'Time', 'Monitor']),
+                             "Normalise to monitor, time or None.")
+        self.declareProperty('Frame', 'Q (sample frame)', StringListValidator(['Q (sample frame)', 'HKL']),
+                             "Selects Q-dimensions of the output workspace")
+        self.declareProperty(FloatArrayProperty("Uproj", [1,0,0], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Defines the first projection vector of the target Q coordinate system in HKL mode")
+        self.declareProperty(FloatArrayProperty("Vproj", [0,1,0], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Defines the second projection vector of the target Q coordinate system in HKL mode")
+        self.declareProperty(FloatArrayProperty("Wproj", [0,0,1], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Defines the third projection vector of the target Q coordinate system in HKL mode")
+        self.declareProperty(FloatArrayProperty("BinningDim0", [-8.02,8.02,401], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Binning parameters for the 0th dimension. Enter it as a"
+                             "comma-separated list of values with the"
+                             "format: 'minimum,maximum,number_of_bins'.")
+        self.declareProperty(FloatArrayProperty("BinningDim1", [-0.82,0.82,41], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Binning parameters for the 1st dimension. Enter it as a"
+                             "comma-separated list of values with the"
+                             "format: 'minimum,maximum,number_of_bins'.")
+        self.declareProperty(FloatArrayProperty("BinningDim2", [-8.02,8.02,401], FloatArrayLengthValidator(3), direction=Direction.Input),
+                             "Binning parameters for the 2nd dimension. Enter it as a"
+                             "comma-separated list of values with the"
+                             "format: 'minimum,maximum,number_of_bins'.")
+        self.declareProperty('KeepTemporaryWorkspaces', False,
+                             "If True the normalization and data workspaces in addition to the normalized data will be outputted")
+        self.declareProperty(WorkspaceProperty("OutputWorkspace", "",
+                                               optional=PropertyMode.Mandatory,
+                                               direction=Direction.Output),
+                             "Output Workspace")
+
+    def validateInputs(self):
+        issues = dict()
+
+        inWS = self.getProperty("InputWorkspace").value
+        d0 = inWS.getDimension(0)
+        d1 = inWS.getDimension(1)
+        d2 = inWS.getDimension(2)
```

Comment:
Bad things happen (Error Report dialog box appears) if you don't have the required number of dimensions.

Reply:
Added check for number of dims